js/services/validation.ts

```diff
@@ -123,11 +123,13 @@ export const validate = (model: any, state: any, flowKey: string): IValidationRe
  * @param flowKey
  */
 export const validateString = (value: string, regex: string | null, message: string, isRequired: boolean, flowKey: string): IValidationResult => {
-    if (isRequired && isValueDefined(value, Component.contentTypes.string)) {
+    if (isRequired && isValueNotDefined(value, Component.contentTypes.string)) {
         return getRequiredResponse(message, flowKey);
     }
 
-    if (!validateRegex(value, regex)) {
+    // Only optional fields get here...
+    // But the regex check should only trigger a validation error if a value has been set
+    if (!validateRegex(value, regex) && !isValueNotDefined(value, Component.contentTypes.string)) {
```

Comment:
This gets more confusing when isValueNotDefined is inverted. Is this value not not defined?

Reply:
Sorted